mongoimport/mongoimport.go

```diff
@@ -502,8 +526,12 @@ func (up *upserter) Insert(doc interface{}) error {
 	var err error
 	if selector == nil {
 		err = up.collection.Insert(document)
-	} else {
+	} else if up.imp.IngestOptions.Mode == modeInsert {
+		// skip if document exists
```

Comment:
If we are in insert mode, the selector must have been nil, so this block is unnecessary. It may make sense to change the else block to an else if that checks if it's in merge mode.